src/storage_persister.rs

```diff
@@ -75,8 +76,59 @@ impl StoragePersister {
         result
     }
 
-    pub fn read_channel_manager(&self, _user_config: UserConfig) {
-        // TODO: Implement
+    #[allow(clippy::too_many_arguments, clippy::type_complexity)]
+    pub fn read_channel_manager<M, T, F, L>(
```

Comment:
I would prefer if this method only handled reading a preexisting `ChannelManager` and if not available, we create a fresh one in another place. Like it is now, we are putting initialisation logic in the `StoragePersister`, which I don't think is intuitive.

Reply:
I feel the same, but other ideas I tried look uglier.
What about we rename the method to `read_or_init_channel_manager()` to make it more explicit?